cf-agent/verify_users_pam.c

```diff
@@ -719,6 +719,19 @@ static bool ChangePassword(const char *puser, const char *password, PasswordForm
     return successful;
 }
 
+static bool IsHashLocked(const char *hash)
+{
+#ifdef HAVE_PW
+    /* Accounts are locked by prepending "*LOCKED*" to the password
+     * hash on FreeBSD and other systems using pw. */
+    return (strstr(hash, "*LOCKED*") != NULL);
```

Comment:
I think this is really OS-specific, not depending on an existence of some utility. Please use `#ifdef __FREEBSD__` here and for the similar checks in this commit.

Reply:
@vpodzime https://github.com/cfengine/core/pull/4002